Use TransitionInfo in more interfaces (more)

wddgit opened the pull request:
#### PR description:

Use the TransitionInfo classes in more interfaces in the Framework. This is the last in a series of pull requests simplifying some interfaces internal to the Framework with the TransitionInfo class. The main goal is to make the code more readable and understandable. It shouldn't affect output or performance. This one is focused on the interfaces from the Worker down into the module base classes.

I am not aware of anything controversial in the one.

#### PR validation:

There should be no change in behavior or output. This relies on existing tests.

wddgit committed:
Use TransitionInfo in more interfaces (more)

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31584/18635

- This PR adds an extra 92KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/interface/stream/ProducingModuleAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/EDAnalyzerAdaptorBase.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc  modified in PR(s): #29553

cmsbuild commented:
A new Pull Request was created by  @wddgit (W. David Dagenhart) for master.

It involves the following packages:

FWCore/Framework


@makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please review it and eventually sign? Thanks.
@makortel this is something you requested to watch as well.
@silviodonato, @dpiparo, @qliphy you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

wddgit commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

makortel commented on FWCore/Framework/interface/stream/EDAnalyzerAdaptor.h:
```diff
@@ -141,13 +142,14 @@ namespace edm {
 
-      void doBeginRun(RunPrincipal const& rp, EventSetupImpl const& ci, ModuleCallingContext const* mcc) final {
+      void doBeginRun(RunTransitionInfo const& info, ModuleCallingContext const* mcc) final {
+        RunPrincipal const& rp = info.principal();
```
How about moving this inside the `if constexpr`? (there are many similar cases below, I won't repeat the comment) I'd expect the compiler to optimize it away in case the condition is false, but I think it would be tiny bit clearer as well.

wddgit replied:
Done. That is better. Thanks. I also did the same thing in the other 7 places this occurs.

makortel commented on FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc:
```diff
@@ -224,13 +226,13 @@ namespace edm {
-                                                                   EventSetupImpl const& ci,
+                                                                   LumiTransitionInfo const& info,
                                                                    ModuleCallingContext const* mcc) {
+      LuminosityBlockPrincipal const& lbp = info.principal();
```
I suppose this line could be omitted and two lines below the `LuminosityBlock`  constructed as
```cpp
      LuminosityBlock lb(info, moduleDescription_, mcc, true);
```

wddgit replied:
Done. Thanks.

makortel commented on FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc:
```diff
@@ -179,43 +180,44 @@ namespace edm {
-                                                       EventSetupImpl const& ci,
+                                                       RunTransitionInfo const& info,
                                                        ModuleCallingContext const* mcc) {
+      RunPrincipal const& rp = info.principal();
```
I suppose this line could be omitted, and two lines below the `Run` be constructed with
```cpp
      Run r(info, moduleDescription_, mcc, true);
```

wddgit replied:
Done. Thanks. (I'll push the new commit in a few minutes)

makortel reviewed (commented):
Looks good, thanks. I had a few minor comments.

cmsbuild commented:
+1
Tested at: 6ca1bcda7e7db134551a4648bb6d4b154709223a
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-732a43/9582/summary.html
CMSSW: CMSSW_11_2_X_2020-09-25-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

wddgit commented:
abort test

cmsbuild commented:
Jenkins tests are aborted.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31584/18640

- This PR adds an extra 92KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/interface/stream/ProducingModuleAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/EDAnalyzerAdaptorBase.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc  modified in PR(s): #29553

cmsbuild commented:
Pull request #31584 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

wddgit commented:
please test

(I just squashed and force pushed after the original submission. I changed nothing other than making the exact changes Matti suggested.)

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 1ef5e5509c2e120f6f9868806aeb5abfe1f62aab
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-732a43/9589/summary.html
CMSSW: CMSSW_11_2_X_2020-09-25-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-732a43/9589/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2539438
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2539415
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

Dr15Jones commented on FWCore/Framework/interface/limited/OutputModuleBase.h:
```diff
@@ -131,34 +130,23 @@ namespace edm {
-                                      ModuleCallingContext const*);
-
-      bool doEvent(EventPrincipal const& ep, EventSetupImpl const& c, ActivityRegistry*, ModuleCallingContext const*);
+      void doStreamBeginRun(StreamID, RunTransitionInfo&, ModuleCallingContext const*);
```
Why is the `RunTransitionInfo&` not `const`? It is for `doBeginRun`?

wddgit replied:
I was copying what was there before. If the RunPrincipal was const, then I made the RunTransitionInfo const. In general, I was doing this in many places in this PR without thinking further.

On closer examination, these four functions here and also four similar functions in global/OutputModuleBase.h are not implemented and not needed for anything. The has_stream_functions template is not specialized for these OutputModuleBase types so these functions are never called. This is just dead code. I will push a commit soon to just delete them.

https://cmssdt.cern.ch/lxr/source/FWCore/Framework/src/WorkerT.cc#0030

Dr15Jones reviewed (commented):
Made it through all the headers.

Dr15Jones reviewed (commented):
Reviewed all files.

wddgit committed:
Delete 8 lines of dead code.

cmsbuild commented:
The code-checks are being triggered in jenkins.

cmsbuild commented:
+code-checks

Logs: https://cmssdt.cern.ch/SDT/code-checks/cms-sw-PR-31584/18683

- This PR adds an extra 20KB to repository

- There are other open Pull requests which might conflict with changes you have proposed:
  - File  FWCore/Framework/interface/stream/EDAnalyzerAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/interface/stream/ProducingModuleAdaptorBase.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/Worker.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/WorkerT.h  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/EDAnalyzerAdaptorBase.cc  modified in PR(s): #29553
  - File  FWCore/Framework/src/stream/ProducingModuleAdaptorBase.cc  modified in PR(s): #29553

cmsbuild commented:
Pull request #31584 was updated. @makortel, @smuzaffar, @cmsbuild, @Dr15Jones can you please check and sign again.

wddgit commented:
please test

All comments should be resolved now.

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
+1
Tested at: 7f792819f69f2bdfe93433e0aa43bc7163fca288
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-732a43/9651/summary.html
CMSSW: CMSSW_11_2_X_2020-09-29-1100
SCRAM_ARCH: slc7_amd64_gcc820

cmsbuild commented:
Comparison job queued.

cmsbuild commented:
Comparison is ready
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-732a43/9651/summary.html

Comparison Summary:
 - No significant changes to the logs found
 - Reco comparison results: 0 differences found in the comparisons
 - DQMHistoTests: Total files compared: 35
 - DQMHistoTests: Total histograms compared: 2542225
 - DQMHistoTests: Total failures: 1
 - DQMHistoTests: Total nulls: 0
 - DQMHistoTests: Total successes: 2542202
 - DQMHistoTests: Total skipped: 22
 - DQMHistoTests: Total Missing objects: 0
 - DQMHistoSizes: Histogram memory added: 0.0 KiB( 34 files compared)
 - Checked 149 log files, 22 edm output root files, 35 DQM output files

makortel commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs (tests are also fine). This pull request will now be reviewed by the release team before it's merged. @silviodonato, @dpiparo, @qliphy (and backports should be raised in the release meeting by the corresponding L2)

silviodonato commented:
+1
